authorization.go

```diff
@@ -19,14 +19,17 @@ type Authorization struct {
 	Amount                     int64                   `json:"amount"`
 	Created                    json.Number             `json:"created"`
 	ReconciliationID           string                  `json:"reconciliation_id"`
-	PaymentMethod              PaymentMethodHref       `json:"payment_method"`
+	PaymentMethod              PaymentMethod           `json:"payment_method"`
```

Comment:
What was wrong with PaymentMethodHref struct?

Reply:
It was an incorrect struct because:
It was 
```golang
 // PaymentMethodHref wraps PaymentMethod with associated href.
type PaymentMethodHref struct {
	Href          string         `json:"href"`
	PaymentMethod *PaymentMethod `json:"payment_method"`
}
```
but according to the documentation and some results in POS we expect the following structure:
```golang
// PaymentMethod is a entity model.
type PaymentMethod struct {
	Href               string            `json:"href"`
	Type               string            `json:"type"`
	TokenType          string            `json:"token_type"`
	PassLuhnValidation bool              `json:"pass_luhn_validation"`
	Token              string            `json:"token"`
	Created            json.Number       `json:"created"`
	Customer           string            `json:"customer"`
	AdditionalDetails  AdditionalDetails `json:"additional_details"`
	BinNumber          json.Number       `json:"bin_number"`
	Vendor             string            `json:"vendor"`
	Issuer             string            `json:"issuer"`
	CardType           string            `json:"card_type"`
	Level              string            `json:"level"`
	CountryCode        string            `json:"country_code"`
	HolderName         string            `json:"holder_name"`
	ExpirationDate     string            `json:"expiration_date"`
	Last4Digits        string            `json:"last_4_digits"`
	IdentityDocument   *IdentityDocument `json:"identity_document"`
	ShippingAddress    *Address          `json:"shipping_address"`
	BillingAddress     *Address          `json:"billing_address"`
	FingerPrint        string            `json:"fingerprint"`
}
```

Link to documentation https://developers.paymentsos.com/docs/apis/payments/1.2.0/#operation/create-an-authorization

![image](https://user-images.githubusercontent.com/1026811/114673410-d2ff9400-9d0e-11eb-854a-f0f61261f013.png)

Reply:
Moving href to the PaymentMethod might be a bit confusing, because it is present only when payment method is returned as an expansion for transaction.
For [retrieve payment](https://developers.paymentsos.com/docs/apis/payments/1.2.0/#operation/retrieve-a-payment) and [create payment](https://developers.paymentsos.com/docs/apis/payments/1.2.0/#operation/create-a-payment) there is no `href` field in response.

Maybe we can fix it by using nested structure instead?
```golang
// PaymentMethodHref wraps PaymentMethod with associated href.
type PaymentMethodHref struct {
	Href          string         `json:"href"`
	PaymentMethod // nested struct
}
```

Reply:
- Hm. I don't see the field  payment_method in the response on the create_payment method, where do you see this?
- For retrieve method I see the field href in response. Look at my screenshot. 
![image](https://user-images.githubusercontent.com/1026811/114725732-6a7fd980-9d45-11eb-819d-2a58d7184484.png)

Also I have checked that in all places ZOOZ returns href attribute in payment_method structure.

Reply:
You are correct, resolved